src/docs/frontend/using-hooks.mdx

```diff
@@ -0,0 +1,73 @@
+---
+title: Using Hooks
+---
+
+To help make components easier to re-use and simpler to understand React
+and the react ecosystem have been trending towards functional components and
+hooks. Hooks are a convenient way to add state and side-effects to functional
+components. They also provide a convenient way for libraries to expose behavior.
+
+While we are generally in favour of hooks we have some recommendations about how
+hooks should be used with the Sentry front-end.
+
+### Using Hooks from Libraries
+
+If a library provides hooks you should use them. Often this will be the only way
+to consume a library. For example `dnd-kit` exposes all of its primitives via
+hooks and we should use the library the way it was intended to be used.
+
+We don't prefer libraries that use or don't use hooks. Instead favour libraries
+with cleaner, simpler APIs, and smaller bundle sizes over ones with larger more
+complex APIs or larger bundle sizes.
+
+### Using React's built-in hooks
+
+The `useState`, `useMemo`, `useCallback`, `useContext`, and `useRef` are welcome
+in any functional component. They are generally a good choice in presentational
+components that need a small amount of state, or access to react primitives like
+references and context. For example, a component with a slide out or expandable
+state.
+
+The `useEffect` hook is more complex and you need to be careful to track your
+dependencies and ensure that subscriptions are cancelled through a cleanup
+callback. Complex chained applications of `useEffect` should be avoided and
+'controller' components should remain class based at this time.
+
+Likewise the `useReducer` hook overlaps with state-management which currently
+has an undecided future. We would like to avoid *yet another* state management
+pattern, so at this time avoid `useReducer`.
+
+### Using context
+
+As we plan our path away from Reflux, the `useContext` hook provide a simpler
```

Comment:
Would `useContext` be preferred to class-based context API React components?

Reply:
My personal bias is to leverage `useContext()` when you need to consume context in a functional component, and not convert to class component or use an HoC to get context pushed in as props.